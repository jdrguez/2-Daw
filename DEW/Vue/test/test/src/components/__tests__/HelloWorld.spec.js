import { mount } from '@vue/test-utils';
import HelloWorld from '../HelloWorld.vue';
import { describe, it, expect } from "vitest";

describe('HelloWorld.vue', () => {
  it('renders correctly with the required prop', () => {
    const wrapper = mount(HelloWorld, {
      props: {
        msg: 'Hello, World!',
      },
    });
    expect(wrapper.find('.greetings').exists()).toBe(true);
    expect(wrapper.find('h1.green').text()).toBe('Hello, World!');
  });

  it('displays the correct static content', () => {
    const wrapper = mount(HelloWorld, {
      props: {
        msg: 'Hello, World!',
      },
    });
    expect(wrapper.html()).toContain("Youâ€™ve successfully created a project with");
    expect(wrapper.html()).toContain("Vite");
    expect(wrapper.html()).toContain("Vue 3");
  });

  it('applies correct styles', () => {
    const wrapper = mount(HelloWorld, {
      props: {
        msg: 'Hello, World!',
      },
    });
    expect(wrapper.find('h1').classes()).toContain('green');
    expect(wrapper.element).toMatchSnapshot();
  });
});
