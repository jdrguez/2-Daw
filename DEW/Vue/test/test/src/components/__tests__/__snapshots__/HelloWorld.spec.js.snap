// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld.vue > applies correct styles 1`] = `
<div
  class="greetings"
  data-v-e17ea971=""
>
  <h1
    class="green"
    data-v-e17ea971=""
  >
    Hello, World!
  </h1>
  <h3
    data-v-e17ea971=""
  >
     Youâ€™ve successfully created a project with 
    <a
      data-v-e17ea971=""
      href="https://vite.dev/"
      rel="noopener"
      target="_blank"
    >
      Vite
    </a>
     + 
    <a
      data-v-e17ea971=""
      href="https://vuejs.org/"
      rel="noopener"
      target="_blank"
    >
      Vue 3
    </a>
    . 
  </h3>
</div>
`;
